File: Backend/src/main/java/ftn/tim16/ClinicalCentreSystem/model/MedicalRecord.java
package ftn.tim16.ClinicalCentreSystem.model;

import javax.persistence.*;
import java.util.HashSet;
import java.util.Set;

@Entity
public class MedicalRecord {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column()
    private Integer height;

    @Column()
    private Integer weight;

    @Column(columnDefinition = "VARCHAR(3)")
    private String bloodType;

    @Column()
    private String alergies;

    @OneToMany(mappedBy = "medicalRecord", fetch = FetchType.LAZY)
    private Set<ExaminationReport> examinationReports = new HashSet<>();

}

File: Frontend/src/app/models/medicalRecord.ts
export class MedicalRecord {
    id: number;
    height: number;
    weight: number;
    bloodType: string;
    allergies: string;

    constructor(id: number, height: number, weight: number, bloodType: string, allergies: string) {
        this.id = id;
        this.height = height;
        this.weight = weight;
        this.bloodType = bloodType;
        this.allergies = allergies;
    }
}